### DISCUSSION TITLE
Make ts aware of the type error

### DISCUSSION THREAD

**Discussion by bluebill1049** [react-hook-form/react-hook-form#5855] [Q&A]
I would like to make type support more strict.

https://codesandbox.io/s/react-hook-form-typescript-forked-b5rdx?file=/src/index.tsx

Any way I can make <Test name="firstName2" /> TS to complain ?

```tsx
import * as React from "react";
import { render } from "react-dom";
import { useForm, FormProvider, FieldValues } from "react-hook-form";
import Test from "./Test";
import "./styles.css";

export type FormData = {
  firstName: string;
  lastName: string;
};

export default function App() {
  const methods = useForm<FormData>();
  const { handleSubmit } = methods;
  const { register } = methods;

  const onSubmit = handleSubmit(({ firstName, lastName }) => {
    console.log(firstName, lastName);
  }); // firstName and lastName will have correct type

  return (
    <FormProvider {...methods}>
      <form onSubmit={onSubmit}>
        {/* input complains fine ? */}
        <input {...register("firstName2")} />
        <Test name="firstName2" /> {/* Any way to make TS complain here ? */}
        <Test name="firstName" />
        <input type="submit" />
      </form>
    </FormProvider>
  );
}

const rootElement = document.getElementById("root");
render(<App />, rootElement);
```

(A question from my dear friend)

**Comment by bluebill1049** [comment]
This is likely not possible, because TS needs to have a context for the form type generic, while you are not passing down any reference to the custom component. TS wouldn't be able to infer the type.

options 

- pass generic at Compoentn `<Component<YourType> />`
- pass `control` down as props 

hope the above answers your question.

**Reply by alifa20** [reply]
Hey there  🤓 , thanks for the reply . I did think of both options but thought there might be any other ways  since `Test` is using `useFormContext<T>` and internally RHF can do some magic  😄

**Reply by bluebill1049** [reply]
Not that I aware, if anyone knows how, let us know.
